pandas/core/frame.py

```diff
@@ -10573,6 +10574,19 @@ def quantile(
                     return res.astype(dtype)
             return res
 
+        has_any_datetime_or_timestamp = any(
+            is_datetime64_any_dtype(x) or is_timedelta64_dtype(x) for x in self.dtypes
+        )
+        if numeric_only and has_any_datetime_or_timestamp:
```

Comment:
We use lib.NO_DEFAULT as default for options we want to check to only raise a warning if numeric only is not supplied. Also I think it is not necessary to check for datetime columns, just raise

Reply:
I guessed that warning everyone would not be wanted. But OK, I will correct it.